Python/ceval.c

```diff
@@ -330,6 +328,15 @@ PyEval_RestoreThread(PyThreadState *tstate)
 #endif
 */
 
+void
+_PyEval_SignalReceived(void)
+{
+    /* bpo-30703: Function called when the C signal handler of Python gets a
+       signal. We cannot queue a callback using Py_AddPendingCall() since this
+       function is not reentrant (uses a lock and a list). */
```

Comment:
you asked me to write "async-signal safe" :-) maybe remove "(uses a lock and a list)"